proposed/0000-Unified-structured-data.md

```diff
@@ -0,0 +1,128 @@
+  smaller than the hash it prevents flooding of a network location.
+- To validate data we confirm signature using hash of (tag_type + version) as nonce. Initial `Put`
+  does not require this signature, but does require the owner contain a value.
+- If previous owners is set then the signature is confirmed using those keys (allows owenr
+  change/transfer etc.). In this case the previous owners is only required on first update and may be
+  remove in next version if the owners are not changed again.
+- To confirm sender of any `Put` (store or overwrite) then we check the signature of sender using same mechanism. For multiple senders we confirm at least 50% of owners have signed the request for `Put`
+
+When `Put` on the network this type is `StructuredData` with a subtype field. The network ignores this subtype
+except for collisions. No two data types with the same name and type can exist on the network.
+
+These types are stored in a disk based storage mechanism such as `Btree` at the NaeManagers (DataManagers) responsible for the area of the network at that `name`.
+
+These types will be limited to 100kB in size (as Immutable Chunks are also limited to 1Mb) which is for the time being a magic number.
+
+If a client requires these be larger than 100kB then the data component will contain a (optionally encrypted) datamap to be able to retrieve chunks of the network.
+
+The network will accept these types if `Put` by a Group and contains a message signed by at least 50% of owners as indicated. For avoidance of doubt 2 owners would require at least 1 have signed, 4 owners would require at least 2 etc. for majority control use an odd number of owners. Any `Put` must obey the mutability rules of these types.
+
+To update such a type the client will `Post` direct (not paying for this again) and the network will overwrite the existing data element if the request is signed by the owner and the version increments. To update a type then there must be an existing type of the same `Identity` and `type` whose owners (or optinally previous owners) includes at least a majority of this new type.
+
+For private data the data filed will be encrypted (at client discretion), for public data this need not be the case as anyone can read that, but only the owner can update it.
+
+##Client perspective 
+
+- Decide on a `type_tag` for a new type.
+- use whichever mechanism to create an `Identity` for this type
+- serialise any structure into `Vec<u8>` and include in data field (can be any structure that is serialisable)
+- store on network via `routing::Put(Identity: location, Data::StructuredData : data, u64: type_tag);`
+- Get from network via `routing::Get(Identity: name, Data::type : type, u64: type_tag);`
+- Mutate on network via `routing::Put(Identity: location, Data::StructuredData : data, u64: type_tag);`
+- Delete from network via `routing::Delete(Identity: name, Data::type : type, u64: type_tag);`
```

Comment:
There will be a `Data` enum in routing which will be like 
```rust
enum Data {
StructuredData(u64),
ImmutableData(u64),
}
``` 
With the `u64` being the type for that data (the tag). For immutableData we have 0,1 & 2 for plain, backup and sacrificial. 

I also think we should have a PlainData type where the content is `Vec<u8>` for folks who use routing as a normal DHT and do not want to be able to set the rules we can here. So our examples may use PlainData to show a simple put/get DHT as we have at the moment, but we can show more complex use cases for sure.